### PR TITLE
Execute beforeClass and afterClass hooks for Cest format

### PR DESCRIPTION
**Pull request by Naktibalda**
I was looking at [Subscriber\BeforeAfterTest](https://github.com/Codeception/Codeception/blob/0355833627f27bcf1b8e995bdd8de20ecaf3bd8b/src/Codeception/Subscriber/BeforeAfterTest.php) for another issue and realized that it wouldn't be too difficult to implement beforeClass and afterClass hooks for Cest format.

It would probably be good to implement attributes too.

Closes https://github.com/Codeception/Codeception/issues/5164

### CODE REVIEW

File: src/Codeception/Subscriber/BeforeAfterTest.php [hunk 1]
```diff
@@ -30,16 +33,31 @@ class BeforeAfterTest implements EventSubscriberInterface
...
+    private ?Actor $actor = null;
+
+    private ?Scenario $scenario = null;
+
     public function beforeClass(SuiteEvent $event): void
     {
         foreach ($event->getSuite()->getTests() as $test) {
-            $testClass = $test::class;
+            if ($test instanceof Cest) {
+                $testClass = $test->getTestClass()::class;
```
Comment (TavoNiievez) [inline src/Codeception/Subscriber/BeforeAfterTest.php:44]: `getTestClass()::class`
https://github.com/Codeception/Codeception/blob/522c7f0f1be5b700ffb8ef8342ba6595f4e16242/src/Codeception/Test/Cest.php#L230-L233
Could we consider renaming that method?

File: tests/cli/ConfigParamsCest.php [hunk 2]
```diff
@@ -6,17 +6,27 @@
 
 final class ConfigParamsCest
 {
-    private function moveToTestDir(CliGuy $I): void
+    /**
+     * @beforeClass
+     */
+    public static function _beforeClass(CliGuy $I)
```
Comment (TavoNiievez) [inline tests/cli/ConfigParamsCest.php:12]: the name of these functions could be anything as long as they include the `@beforeClass` annotation, right?
why are they `static`?

Comment (Naktibalda) [reply tests/cli/ConfigParamsCest.php:12]: PHPUnit convention - only static methods can be executed before and after class.

We would have to replace `TestUtil::getHookMethods($testClass);` with our own code, which is actually a good idea.

Comment (Naktibalda) [reply tests/cli/ConfigParamsCest.php:12]: I will try to replace TestUtil with Codeception annotation parsing code.

**Review by DavertMik — APPROVED** [review]

**Comment by Naktibalda** [comment]
I have no time to finish this PR now, so I will leave it for 5.1.0.

I think that Cest format must support `#[\Codeception\Attribute\BeforeClass]` on non-static methods instead of relying on PHPUnit's annotation parser/attributes.

File: src/Codeception/Subscriber/BeforeAfterTest.php (hunk 1 above)
Comment (Naktibalda) [reply src/Codeception/Subscriber/BeforeAfterTest.php]: Renamed

File: tests/cli/ConfigParamsCest.php (hunk 2 above)
Comment (Naktibalda) [reply tests/cli/ConfigParamsCest.php]: instance methods are supported now

**Comment by Naktibalda** [comment]
Interesting, test don't fail for me locally.

**Comment by Naktibalda** [comment]
I moved code cleanup to https://github.com/Codeception/Codeception/pull/6499

File: tests/support/CliHelper.php [hunk 3]
```diff
@@ -20,7 +20,7 @@ public function _after(\Codeception\TestInterface $test)
     {
         codecept_debug('deleting dirs');
         $this->getModule('Filesystem')->deleteDir(codecept_data_dir() . 'sandbox');
-        chdir(\Codeception\Configuration::projectDir());
+        $this->getModule('Filesystem')->amInPath(codecept_root_dir());
```
Comment (Naktibalda) [inline tests/support/CliHelper.php:23]: This change broke some unit tests that were relying on matching relative tests when `unit` suite is executed after `cli`

Comment (Naktibalda) [reply tests/support/CliHelper.php:23]: Actually the problem was that `amInPath` called in `ConfigParamsCest::_beforeClass` changed working directory for all tests.

**Comment by Naktibalda** [comment]
Maybe later.

**Pull request closed without merging**